src/with-urql-client.tsx

```diff
@@ -77,7 +72,7 @@ function withUrqlClient<T = any, IP = any>(
       );
     };
 
-    withUrql.getInitialProps = async (ctx: NextContextWithAppTree) => {
+    withUrql.getInitialProps = async (ctx: NextUrqlContext) => {
```

Comment:
So this type should actually be just:

```suggestion
    withUrql.getInitialProps = async (ctx: NextContext) => {
```

the reason being that, at this point in the function's execution, we can't safely access `ctx.urqlClient` b/c it doesn't exist yet (it gets added on L83).

Ofc, that'll result in a type error, so on L83 you'll need to do:

```typescript
(ctx as NextUrqlContext).urqlClient = urqlClient;
```

It's a small thing, but I think it better signals to future readers what we're doing here, which is actually shoving a property onto the original `ctx` object so that components lower in the tree can access it.

Reply:
Ohh I see, makes sense. Would it also be helpful to coerce `ctx` as `NextUrqlContext` when we pass it to `Page.getInitialProps` on L89?

Reply:
Ah yep, good catch – that way if a user wants to access `ctx.urqlClient` in their own `getInitialProps` function they can. Let's do that, and then I'll merge this!

Reply:
Yup, and I figure it's at least more accurate! Added that in 👍